source/core/src/main/com/csse3200/game/components/player/InventoryComponent.java

```diff
@@ -231,76 +451,41 @@ public int getHeldIndex() {
    * @return integer representation of count
    */
   public int getItemCount(Entity item) {
-    return itemCount.getOrDefault(item, 0);
-  }
-
-  /**
-   * Returns the position of an item
-   * @param item entity that is the item we want to find
-   * @return Point of the positional representation of the inventory
-   */
-  public Point getItemPosition(Entity item) {
-    return itemPosition.get(item);
+    return this.itemCount.getOrDefault(item.getComponent(ItemComponent.class).getItemName(), 0);
   }
 
-  /**
-   * Returns a boolean value representing whether or not an item is at a point representation of the inventory
-   * @param point Point to check whether an item is at that position
-   * @return Boolean representing whether or not item is at point
-   */
-  public Boolean getItemAtPoint( Point point) {
-    for (Map.Entry<Entity, Point> entry : itemPosition.entrySet()) {
-      if (entry.getValue().equals(point)) {
-        return true;
-      }
-    }
-    return false;
+//  @Override
+//  public void write(Json json) {
+//    json.writeObjectStart(this.getClass().getSimpleName());
+//    json.writeObjectStart("inventory");
+//    for (Entity e : inventory) {
+//      json.writeObjectStart("item");
+//      e.writeItem(json);
+//      json.writeValue("count", getItemCount(e));
+//      json.writeValue("X", getItemPosition(e).x);
+//      json.writeValue("Y", getItemPosition(e).y);
+//      json.writeObjectEnd();
+//    }
+//    json.writeObjectEnd();
+//    json.writeObjectEnd();
+//  }
+
+  public Integer getItemCount(int i) {
+    return this.itemCount.getOrDefault(this.itemPlace.get(i), 0);
   }
 
   /**
-   * Get item at a point
-   * @param point Point representation of inventory position
-   * @return Item at that position or null
+   * Function to trigger the update of ItemSlot at the pass index.
+   * @param index
    */
-  public Entity getItem(Point point) {
-    for (Map.Entry<Entity, Point> entry : itemPosition.entrySet()) {
-      if (entry.getValue().equals(point)) {
-        return entry.getKey();
-      }
-    }
-    return null;
-  }
-
 
-  /**
-   * Set point position for item
-   * @param item item to set position of
-   * @param point position to set
-   */
-  public void setItemPosition(Entity item, Point point) {
-    itemPosition.put(item, point);
-  }
+  public void itemChange(int index) {
+    if(index >= 0 && index < 10) {
+      // Only update toolbar
 
+    } else {
+      // update both toolbar and inventory
```

Comment:
Does nothing at this point, and you even if the index is less than 10, you still need to update both. Function doesn't make sense or have any tangible purpose.

Reply:
The function was to trigger updates of specific itemSlot but I think you took care of it in the display itself.

Removing the function.